app.js

```diff
@@ -99,6 +100,25 @@ app.use('/interactive', (req, res) => {
     user: { id: sender },
     // much more avb inside, check documentation
   } = parsed
+  if (callback_id === 'demo') {
+    // needed for modal submission
+    res.status(200).json({ 'response_action': 'clear' })
+
+    const {
+      date: { datepicker: { selected_date: date } },
+      startTime: { 'timepicker-start': { selected_time: start } },
+      endTime: { 'timepicker-end': { selected_time: end} }
+    } = values
+
+    gCalendarCreateEvent({ date, start, end }).then((status) => {
+      return axios.post(private_metadata, {
+        response_type: 'ephemeral',
+        text: status === 200
+          ? 'Event added to the Demo calendar'
+          : 'COULD NOT save this event. Double check the time selected'
+      })
+    })
```

Comment:
this is not doing early return unless `return` is also done before `gCalendarCreateEvent`

Reply:
on top of that, since the implementation `gCalendarCreateEvent` is implemented in a way such that any errors aren't propagated out, this would create unexpected behaviour when the `axios.post` operation throws an error.

i would advise against blanket try/catch. see below in `gCalendarCreateEvent` for context

Reply:
what do you mean by early return?

should #56 also get rid of try and catch for the same reasons, bubble up in diff?

Reply:
early return means that a block of logic gets "early" exit from its outer scope altogether. typically seen in imperative flow control such as `if/else` or `while/for`.

```js
function someFn(x) {
  if (typeof x === 'number') {
    return x + 5 // early return/exit from the function
  }
  // ... rest of the logic
}
```

the nature of promise interface is more "declarative" than "imperative", so the return itself needs to happen also at its "root" or the rest of the outer-scope code _may_ still execute:

```js
if (command == '...') {
  somePromise().then(...).catch(...)
}
if (command == '...) {
  ...
}
... // <- if not excluded from any of the above `if` imperatives, any code would run after the promise since promise is async
```